addons/website/models/website.py

```diff
@@ -656,6 +656,30 @@ def is_user(self):
     def is_public_user(self):
         return request.env.user.id == request.website.user_id.id
 
+    @api.model
+    def viewsref(self, keys, raise_if_not_found=True):
+        View = self.env['ir.ui.view']
+        initial_order = {key: View for key in keys}
+        if 'website_id' in self._context:
+            domain = [('key', 'in', keys)] + self.env['website'].website_domain(self._context.get('website_id'))
+            order = 'website_id'
+        else:
+            domain = [('key', 'in', keys)]
+            order = View._order
+        views = View.with_context(active_test=False).search(domain, order=order)
+        views = views.filter_duplicate()
+
+        if raise_if_not_found and len(keys) != len(views):
+            raise ValueError("One of the requested views couldn't be found.")
+
+        # preserve given order
+        res = View
```

Comment:
jke - tocheck

Reply:
Something to be done here?

Reply:
I'd remove that one for the PR atm as it is not used, we still need to think more deeply about a cache on view get methods etc, and this version is not fully working actually